buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTestingPlugin.kt

```diff
@@ -103,18 +103,23 @@ class DistributionTestingPlugin : Plugin<Project> {
     fun DistributionTest.setSystemPropertiesOfTestJVM(project: Project) {
         // use -PtestVersions=all or -PtestVersions=1.2,1.3…
         val integTestVersionsSysProp = "org.gradle.integtest.versions"
+        val sysPropValue = System.getProperty(integTestVersionsSysProp)
         if (project.hasProperty("testVersions")) {
```

Comment:
I think there is still some left over cruft from when we had 2 different system properties controlling which versions to use in a test.  We used to have one property that would control which set of versions to test and another which would control the list of versions. That's sort of what this change is building back. Could we just simplify this down to one property?

This whole method would become:
```
val integTestVersionsSysProp = "org.gradle.integtest.versions"
if (project.hasProperty("testVersions")) {
    systemProperties[integTestVersionsSysProp] = project.property("testVersions")
} else {
    systemProperties[integTestVersionsSysProp] = "default"
}
```
In places where we set testPartialVersions to true, we would set testVersions to partial.
Where we set testAllVersions to true, we would set testVersions to all.

We'd ignore someone trying to set the system property org.gradle.integtest.versions directly.

Reply:
Interesting, since passing `-PtestAllVersions=true` or `-PtestPartialVersions=true` on the CLI never worked, it can be simplified indeed so that only `-PtestVersions=...` is supported.

Will do that then!